apps/src/templates/teacherNavigation/PageHeader.tsx

```diff
@@ -28,7 +34,24 @@ const PageHeader: React.FC = () => {
   const isLoadingSectionData = useAppSelector(
     state => state.teacherSections.isLoadingSectionData
   );
+  const [ageGatedModalOpen, setAgeGatedModalOpen] = useState(false);
+  const toggleAgeGatedModal = () => {
+    setAgeGatedModalOpen(!ageGatedModalOpen);
+  };
   const selectedSection = useAppSelector(selectedSectionSelector);
+  const dispatch = useAppDispatch();
+  useEffect(() => {
+    dispatch(loadSectionStudentData(selectedSection.id));
+  }, [dispatch, selectedSection?.id]);
+  const studentData = useAppSelector(
+    state => state.manageStudents?.studentData
+  );
+  const ageGatedStudents = filterAgeGatedStudents(
+    convertStudentDataToArray(studentData)
+  );
+  const ageGatedStudentsUsState =
+    ageGatedStudents?.length > 0 ? ageGatedStudents[0].usState : undefined;
```

Comment:
Can you just use `ageGatedStudents?.length > 0 && ageGatedStudents[0].usState`?

Reply:
I could, but I don't like mixing types. In that case, ageGatedStudentsUsState could be `undefined`, `string`, or `boolean`. Am I being too conservative?

Reply:
Fair point!